internal/unboundedchan/unboundedchan.go

```diff
@@ -0,0 +1,82 @@
+package unboundedchan
```

Comment:
I'm very against adding unbounded chan, I think it solves no problem and will make debugging harder. What happens when we run out of memory because an unbounded chan is so full of crap that everything stalls? How do we know what happened?
So then we should add some monitoring so we get a warning if the unbounded chan is too full? Well why not use a bounded chan at that point. 

It's also more code we need to maintain that may contain bugs.